File: MainProject/Code.Domains/Models/ValueObjects/PlanetGenerates/ChunkLod.cs
namespace Domains.Models.ValueObjects.PlanetGenerates;

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH
/// Date: 2025-04-11 19:09:11
public enum ChunkLod
{
    JustHex, // 每个地块只有六个平均高度点组成的六边形（非平面）
    PlaneHex, // 高度立面，无特征，无河流的六边形
    SimpleHex, // 最简单的 Solid + 斜面六边形 
    TerracesHex, // 增加台阶
    Full, // 增加边细分
}

File: TerraObserver/TO.FSharp/Domains/Components/HexSpheres/Chunks/ChunkLod.fs
namespace TO.FSharp.Domains.Components.HexSpheres.Chunks

open Friflo.Engine.ECS
open Godot.Abstractions.Extensions.Chunks

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH (ZeromaXHe)
/// Date: 2025-06-08 13:30:08
[<Struct>]
type ChunkLod =
    interface IComponent
    val Lod: ChunkLodEnum
    new(lod) = { Lod = lod }
